core/src/main/java/nl/nn/adapterframework/http/rest/ApiListenerServlet.java

```diff
@@ -444,7 +445,15 @@ else if(segment.startsWith("{") && segment.endsWith("}")) {
 			if (!ServletFileUpload.isMultipartContent(request)) {
 				body = Misc.streamToString(request.getInputStream(),"\n",false);
 			}
-			//TODO: String correlationId = request.getHeader("message-id");
+
+			String messageId = null;
+			if(StringUtils.isNotEmpty(listener.getMessageIdHeader())) {
+				String messageIdHeader = request.getHeader(listener.getMessageIdHeader());
+				if(StringUtils.isNotEmpty(messageIdHeader)) {
+					messageId = messageIdHeader;
+				}
+			}
+			PipeLineSessionBase.setListenerParameters(messageContext, messageId, null, null, null); //We're only using this method to keep setting id/cid/tcid uniform
```

Comment:
Dit moet m.i. niet hier, maar in ApiListener.getIdFromRawMessage()

Reply:
Ik heb hier issue #769 voor aangemaakt